admin/src/components/Title.js

```diff
@@ -1,61 +1,101 @@
 import { onChangeFieldHandler } from "./PresetsTableRow";
 import { PresetsManager } from "./PresetsManager";
 import { usePresetWidgetService } from "../services/presetWidget";
+import { useTitleWidgetService } from "../services/titleWidget";
 
-const parseJSONOrDefault = (text, defult) => {
-    try {
-        return JSON.parse(text);
-    } catch (e) {
-        return defult;
-    }
+const PreviewSVGDialog = ({ open, handleClose, id }) => {
+    const { enqueueSnackbar } = useSnackbar();
+    const service = useTitleWidgetService("/title", errorHandlerWithSnackbar(enqueueSnackbar), false);
+    const [content, setContent] = useState(undefined);
+    useEffect(() => open && service.getPreview(id).then(r => setContent(r.content)), [open, id]);
+    return (
+        <Dialog onClose={handleClose} open={open} fullWidth maxWidth="md">
+            <DialogTitle>Title preview</DialogTitle>
+            <DialogContent>
+                <Card sx={{ borderRadius: 0 }}>
+                    {content === undefined &&
+                        <Stack alignItems="center" sx={{ py: 3 }}><CircularProgress/></Stack>}
+                    {content !== undefined && <object type="image/svg+xml" data={content} style={{ width: "100%" }}/>}
+                </Card>
+            </DialogContent>
+            <DialogActions>
+                <Button onClick={handleClose} autoFocus>OK</Button>
+            </DialogActions>
+        </Dialog>
+    );
+};
+PreviewSVGDialog.propTypes = {
+    open: PropTypes.bool.isRequired,
+    handleClose: PropTypes.func.isRequired,
+    id: PropTypes.number.isRequired,
 };
 
-const ParamsLine = ({ pKey, pValue }) => (<Box sx={{}}><b>{pKey}</b>: {JSON.stringify(pValue)}</Box>);
+const TemplateEditor = ({ value, onSubmitAction, onChangeHandler }) => {
+    const [templates, setTemplates] = useState([]);
 
-ParamsLine.propTypes = {
-    pKey: PropTypes.string.isRequired,
-    pValue: PropTypes.any.isRequired,
+    const service = useTitleWidgetService("/title", undefined, false);
+    useEffect(() => service.getTemplates().then(ts => setTemplates(ts)), []);
+
+    return (<Box onSubmit={onSubmitAction} component="form" type="submit">
+        <Autocomplete
+            disablePortal
+            freeSolo
+            sx={{ width: 1 }}
+            size="small"
+            value={value}
+            onChange={(_, value) => onChangeHandler(value) }
+            options={templates}
+            renderInput={(params) => <TextField {...params} label="SVG preset"/>}/>
+    </Box>);
 };
+TemplateEditor.propTypes = ValueEditorPropTypes;
+
+const ParamsLine = ({ pKey, pValue }) => (<Box sx={{}}><b>{pKey}</b>: {pValue}</Box>);
+ParamsLine.propTypes = { pKey: PropTypes.string.isRequired, pValue: PropTypes.any.isRequired };
+
+const parseParamsData = input =>
+    input.split("\n")
+        .map(line => line.split(/:\W?/, 2))
+        .filter(r => r.length >= 2)
+        .reduce((ac, [ k, v ]) => { ac[k] = v; return ac; }, {});
```

Comment:
If you want more of an FP-approach, use this instead:
```javascript 
.reduce((ac, [ k, v ]) => {...ac, [k]: v}, {});
```
Your reducer is a bit faster because it does not copy an object and instead modifies the original one, but, it's not pure.